File: examples/30/prefix_sum/prefix_sum.py
from time import time_ns

list = [1] * 10_000   # a Python list
# print(list) # for 10 elements: [1, 1, 1, 1, 1, 1, 1, 1, 1, 1]
tik = time_ns()
for i in range(1, len(list)):
    list[i] += list[i - 1]
tok = time_ns()
# print(list) # for 10 elements: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
print(f"Time spent per element: {(tok - tik) / len(list)} ns")
# => Time spent per element: 67.5547 ns

File: examples/30/prefix_sum/prefix_sum.rs
use std::time::Instant;

fn main() {
    let mut list: Vec<i64> = vec!();
    for _ in 0..10_000 {
        list.push(1)
    }
    let tik = Instant::now();
    for i in 1..list.len() {
        list[i] += list[i - 1]
    }
    let tok = Instant::now();
    println!("Time spent per element: {:?} ns",
        ((tok - tik).as_nanos() as f64) / list.len() as f64)
}

// => Time spent per element: 9.752 ns
// rustc -O: Time spent per element: 2.488 ns